### PR TITLE
[swift-ide-test] Use same code completion extraction logic for batch code completion and single code completion

### PR DESCRIPTION
**Pull request by ahoppen**
If the test case has a code completion token of the form `#^TOKEN?check=CHECK^#` from batch code completion, this allows us to run `swift-ide-test` with `--code-completion-token TOKEN` instead of `--code-completion-token 'TOKEN?check=CHECK'`, which is more convenient.

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

File: tools/swift-ide-test/swift-ide-test.cpp [hunk 1]
```diff
@@ -904,10 +1062,30 @@ static bool performWithCompletionLikeOperationParams(
...
+    }
+  }
+
+  auto CompletionBuffer = ide::makeCodeCompletionMemoryBuffer(CleanFile.get(), Offset, CleanFile->getBufferIdentifier());
+
+
+//  llvm::errs() << OldCleanFile.get()->getBuffer();
+//  llvm::errs() << "\n\n\n-------\n\n\n";
+//  llvm::errs() << CleanFile.get()->getBuffer() << '\n';
+//  auto FilesEqual = OldCleanFile.get()->getBuffer() == CleanFile.get()->getBuffer();
```
Comment (rintaro) [inline tools/swift-ide-test/swift-ide-test.cpp:1087]: Could you remove this?

File: tools/swift-ide-test/swift-ide-test.cpp [hunk 2]
```diff
@@ -904,10 +1062,30 @@ static bool performWithCompletionLikeOperationParams(
...
+
+  unsigned Offset = ~0U;
+  for (auto Token : Tokens) {
+    if (Token.Name == CodeCompletionToken) {
+      Offset = Token.Offset;
+      break;
+    }
+  }
+
+  auto CompletionBuffer = ide::makeCodeCompletionMemoryBuffer(CleanFile.get(), Offset, CleanFile->getBufferIdentifier());
```
Comment (rintaro) [inline tools/swift-ide-test/swift-ide-test.cpp:1081]: It doesn't really matter, but this should be after `if (Offset == ~0U)` check.

**Review by rintaro — APPROVED** [review]
Assuming you just moved `CompletionTestToken` without any modification, this LGTM.

I guess another solution would be making non-batch `-code-completion` just another mode of `doBatchCodeCompletion` where it only runs the single token and emits the result to stdout. But this PR itself looks good.

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

**Pull request merged**